dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/relationship/RelationshipOperationParamsMapper.java

```diff
@@ -0,0 +1,74 @@
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.export.relationship;
+
+import static org.hisp.dhis.tracker.export.relationship.RelationshipQueryParams.EMPTY;
+
+import lombok.RequiredArgsConstructor;
+import org.hisp.dhis.feedback.ForbiddenException;
+import org.hisp.dhis.feedback.NotFoundException;
+import org.hisp.dhis.tracker.export.enrollment.EnrollmentParams;
+import org.hisp.dhis.tracker.export.enrollment.EnrollmentService;
+import org.hisp.dhis.tracker.export.event.EventParams;
+import org.hisp.dhis.tracker.export.event.EventService;
+import org.hisp.dhis.tracker.export.trackedentity.TrackedEntityParams;
+import org.hisp.dhis.tracker.export.trackedentity.TrackedEntityService;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Maps {@link RelationshipOperationParams} to {@link RelationshipQueryParams} which is used to
+ * fetch enrollments from the DB.
+ */
+@Component
+@RequiredArgsConstructor
+class RelationshipOperationParamsMapper {
+
+  private final TrackedEntityService trackedEntityService;
+
+  private final EnrollmentService enrollmentService;
+
+  private final EventService eventService;
+
+  @Transactional(readOnly = true)
+  public RelationshipQueryParams map(RelationshipOperationParams params) throws NotFoundException {
+    try {
+      return new RelationshipQueryParams(
+          switch (params.getType()) {
+            case TRACKED_ENTITY -> trackedEntityService.getTrackedEntity(
+                params.getIdentifier(), TrackedEntityParams.TRUE, true);
+            case ENROLLMENT -> enrollmentService.getEnrollment(
+                params.getIdentifier(), EnrollmentParams.TRUE, true);
+            case EVENT -> eventService.getEvent(params.getIdentifier(), EventParams.TRUE);
+            case RELATIONSHIP -> null;
+          });
+    } catch (ForbiddenException ex) {
+      return EMPTY;
```

Comment:
Why are you catching this?

Reply:
To maintain the same behaviour as we have now.
This check will be repeated when checking the access for the relationship itself and the inaccessible ones will just be filtered out.
I am in a discussion with @JoakimSM and @Bekkalizer to check if this behaviour make sense, but anyway it will be tackled in a future PR